Rules/UseUsingScopeModifierInNewRunspaces.cs

```diff
@@ -0,0 +1,375 @@
+        /// <returns>The description of this rule</returns>
+        public string GetDescription()
+        {
+            return string.Format(CultureInfo.CurrentCulture, Strings.UseUsingScopeModifierInNewRunspacesDescription);
+        }
+
+        /// <summary>
+        /// GetSourceType: Retrieves the type of the rule: builtin, managed or module.
+        /// </summary>
+        public SourceType GetSourceType()
+        {
+            return SourceType.Builtin;
+        }
+
+        /// <summary>
+        /// GetSeverity: Retrieves the severity of the rule: error, warning of information.
+        /// </summary>
+        /// <returns></returns>
+        public RuleSeverity GetSeverity()
+        {
+            return RuleSeverity.Warning;
+        }
+
+        /// <summary>
+        /// GetSourceName: Retrieves the module/assembly name the rule is from.
+        /// </summary>
+        public string GetSourceName()
+        {
+            return string.Format(CultureInfo.CurrentCulture, Strings.SourceName);
+        }
+
+        /// <summary>
+        /// Checks if a variable is initialized and referenced in either its assignment or children scopes
+        /// </summary>
+        /// <param name="scriptBlockAst">Ast of type ScriptBlock</param>
+        /// <param name="fileName">Name of file containing the ast</param>
+        /// <returns>An enumerable containing diagnostic records</returns>
+        private IEnumerable<DiagnosticRecord> AnalyzeScriptBlockAst(ScriptBlockAst scriptBlockAst, string fileName)
+        {
+            var astsToProcess = new List<Ast>();
+            var nonAssignedNonUsingVars = new List<VariableExpressionAst>();
+
+            astsToProcess.AddRange(
+                FindAllInlineScriptAsts(scriptBlockAst));
+
+            astsToProcess.AddRange(
+                FindAllStartJobAsts(scriptBlockAst));
+
+            astsToProcess.AddRange(
+                FindAllForeachParallelAsts(scriptBlockAst));
+
+            //   If -ComputerName or -Session is not specified, you cannot use a variable with a using: scope modifier. 
+            //   Also tricky; one can open a persistent session, and use subsequent `invoke-command` calls to that, where variables
+            //   assigned in previous calls are still valid:
+            //   https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_remote_variables?view=powershell-7#using-remote-variables
+            //   Because -ComputerName and -Session parameters are in different parameter sets, we can treat them separately
+            astsToProcess.AddRange(
+                FindAllInvokeCommandComputerAsts(scriptBlockAst));
+
+            // process Invoke-Command -Session separately
+            nonAssignedNonUsingVars.AddRange(
+                ProcessInvokeCommandSessionAsts(scriptBlockAst));
+
+
+            foreach (var ast in astsToProcess)
+            {
+                if (ast == null)
+                    continue;

```

Comment:
Please use braces for conditional statements

Reply:
I have fixed this throughout, thanks